kite-data/kite-data-hive/src/main/java/org/kitesdk/data/spi/hive/HiveExternalMetadataProvider.java

```diff
@@ -82,6 +83,25 @@ public DatasetDescriptor create(String namespace, String name, DatasetDescriptor
           .build();
     }
 
+    if (descriptor.getSchemaUrl() == null) {
+
+      // No schema URL is provided, so create one based on the dataset.
+      Path managerPath = new Path(new Path(newDescriptor.getLocation()),
+          SCHEMA_DIRECTORY);
```

Comment:
Shouldn't this always create a SchemaManager? If the URL is in the manager then that's a good thing (is that case detected?). Otherwise, the right thing to do is to copy the Schema to a managed location and rely on that one. That addresses the concern that the schema is changed without going through Kite (since Kite currently has just a pointer).